remote_llm.py

```diff
@@ -3,14 +3,16 @@
 import os
 from ast import List
 from logging import getLogger
+from typing import Dict
+
 
 # Third-party imports.
-from requests import post, get
+import requests
 from requests.exceptions import HTTPError
 
 log = getLogger(__name__)
 
-def generate_text(preprompt, prompt, model):
+def generate_text(idempotency_uuid, preprompt, prompt, model, history= []):
```

Comment:
Dont use mutables in python arguments 

https://docs.python-guide.org/writing/gotchas/
```suggestion
def generate_text(idempotency_uuid, preprompt, prompt, model, history= None):
  if history is None:
    history = []
```

Reply:
Python is a foreign land!